xdsl/interpreters/experimental/eqsat_pdl.py

```diff
@@ -0,0 +1,260 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import IO, Any, cast
+
+from xdsl.context import MLContext
+from xdsl.dialects import eqsat, pdl
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.interpreter import Interpreter, InterpreterFunctions, impl, register_impls
+from xdsl.interpreters.experimental.pdl import PDLMatcher
+from xdsl.ir import Attribute, Operation, OpResult, SSAValue, TypeAttribute
+from xdsl.irdl import IRDLOperation
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.rewriter import InsertPoint
+from xdsl.transforms.convert_onnx_to_linalg import get_root_op
+from xdsl.utils.exceptions import InterpretationError
+
+
+@dataclass
+class EqsatPDLMatcher(PDLMatcher):
+    def match_operand(
+        self, ssa_val: SSAValue, pdl_op: pdl.OperandOp, xdsl_val: SSAValue
+    ):
+        owner = xdsl_val.owner
+        assert isinstance(owner, eqsat.EClassOp)
+        assert (
```

Comment:
I guess this will no longer be true after the first rewrite, i.e. an eclass may have more than one operand?